packages/node_modules/pouchdb-adapter-leveldb/src/migrate.js

```diff
@@ -36,7 +36,7 @@ var doMigrationOne = function (name, db, callback) {
       };
     }
     var sub = db.sublevel(store, opts);
-    var orig = levelup(storePath, opts);
+    var orig = level(storePath, opts);
```

Comment:
I tried doing

```js
var orig = levelup(leveldown(storePath), opts);
```

But then I got the following error:

![levelup-leveldown-error-migrate](https://user-images.githubusercontent.com/308049/34066464-46c39c9c-e20f-11e7-9e6f-39398c2f228f.png)

This is a bit trial and error and would really like to know _why_ that doesn't work. I assume it has to do with encodings.

cc @daleharvey @vweevers

Reply:
I *think* the code here uses two db instances. One is wrapped with sublevel (which handles encodings), one is plain levelup. Both are expected to support `binary` and `json` encoding.

The levelup instance doesn't, and that's probably why the value `metadata` here isn't an object:

https://github.com/pouchdb/pouchdb/blob/e8d8f79a776cfb7196344f80310d09935278834d/packages/node_modules/pouchdb-merge/src/winningRev.js#L10

Hence `Cannot read property 'slice' of undefined`.

So I think it also needs `encoding-down` @ralphtheninja.

Reply:
@daleharvey probably best if you do use `levelup` rather than `level` here, to avoid version mismatches.